### ISSUE TITLE
Issues with running on Mac OS

### ISSUE THREAD

**Issue by Aces-and-Jacks** [ugackMiner53/DiscordProximity#4]
1.19.2

https://mclo.gs/4yWmgmd

**Comment by ugackMiner53** [comment]
Seems to be a library issue (see issues [#54](https://github.com/JnCrMx/discord-game-sdk4j/issues/54) [#48](https://github.com/JnCrMx/discord-game-sdk4j/issues/48))

I'll look into MacOS support, but given that I only run Linux and Windows, debugging is going to be a bit challenging...

**Comment by Aces-and-Jacks** [comment]
yeah I looked on there as well. Saw there was a PR to fix it but hasn't been merged there yet. Figured I'd try anyway

**Comment by eklyst** [comment]
did you ever figure this one out?

**Comment by Aces-and-Jacks** [comment]
> did you ever figure this one out?

It's a library issue that this mod relies on. Issue 48 seems where the most development on apple silicon was

**Comment by eklyst** [comment]
so is there a solution to this?

**Comment by ugackMiner53** [comment]
Not currently, sorry.

**Comment by letorbi** [comment]
A solution might be to use the *java-imp*` branch of *discord-game-sdk4j*, which provides pure Java bindings for the Discord GameSDK. It is currently known to work with Windows, Mac (Intel & ARM) as well as Arch Linux (and probably other distros).

The branch is mostly API compatible to the master branch and even though some things might be missing, the main features are available.

I am using that branch for [a little project of mine](https://github.com/letorbi/discord-rich-presence-for-processing), and it is working fine so far. You can [use JitPack to include the java-impl branch](https://jitpack.io/#JnCrMx/discord-game-sdk4j/java-impl-SNAPSHOT) into your project.

**Comment by ugackMiner53** [comment]
Unfortunately, the Discord Game SDK has removed support for Voice and Lobbies, so now *no library will work*.

See the newly made #11 for more details.
